azurelinuxagent/common/event.py

```diff
@@ -430,10 +454,14 @@ def add_event(name, op=WALAEventOperation.Unknown, is_success=True, duration=0,
 
 def add_log_event(level, message, reporter=__event_logger__):
     if reporter.event_dir is None:
+        # See #1035 for not adding warning statements here.
```

Comment:
can you expand on what should not be done and why? thanks

Reply:
I'll add the following description in the code as well - 

The issue with 1035 was never-ending machinery generating events continuously and stretching the system.

At t=0 - Assume 999 events are already in the events folder.
At t=1 - Now, assuming that a new log.WARN [1] event comes in and needs to be written - when saving this event on the disk `save_event` would generate another log.WARN [2] as it warns that there are too many events on the dist.

Thus, this new WARN [2] event again would propagate more WARN events to be written, thus introducing a never-ending cycle of WARNs being written.

Thus to prevent the chance of this happening, I introduced periodic_WARNs. Thus curtailing to one event needed to be written at most, and within the next minute, it would be cleared away.